src/service.rs

```diff
@@ -235,6 +237,17 @@ mod tests {
         fn document_highlight(&self, _: TextDocumentPositionParams) -> Self::HighlightFuture {
             Box::new(future::ok(None))
         }
+
+        fn signature_help(&self, _: TextDocumentPositionParams) -> Self::SignatureHelpFuture {
+            Box::new(future::ok(None))
+        }
+
+        fn goto_implementation(
```

Comment:
This method should be moved to immediately below `LspService::goto_type_definition()`.